src/context/apiClient/apiClientContextController/apiError/apiError.ts

```diff
@@ -1,31 +1,49 @@
 import { AxiosError } from 'axios';
 import zod from 'zod';
 
-import { ApiError, BasicErrorData, FormErrorData } from './apiError.types';
+import { BasicApiError, BasicErrorData, FormApiError, FormErrorData, UnknownApiError } from './apiError.types';
 
-export const getStandardizedApiError = (error: AxiosError<unknown>): ApiError => {
+export class ApiError<T extends BasicApiError | FormApiError | UnknownApiError> extends Error {
+  originalError: T['originalError'];
+  statusCode: T['statusCode'];
+  type: T['type'];
+  data: T['data'];
```

Comment:
```suggestion
  readonly originalError;
  readonly statusCode;
  readonly type;
  readonly data;
```
I think we can simplify it like that, cause types will be inferred based on the assignment in the constructor

Reply:
Yeah it's a leftover as I had some issues with types when setting this up. I was figuring out whether it's possible to achieve proper type inference with just type `ApiError<BasicApiError | FormApiError | UnknownApiError>`